src/Tribe/PUE/Checker.php

```diff
@@ -1538,43 +1582,46 @@ public function get_state( $force_recheck = false ) {
 		public function update_state( $value ) {
 
 			update_site_option( $this->pue_option_name, $value );
-
 		}
 
 		/**
 		 * Check for plugin updates.
 		 *
 		 * The results are stored in the DB option specified in $pue_option_name.
 		 *
-		 * @param array   $updates
-		 * @param boolean $force_recheck
+		 * @param array|object $updates       Existing updates.
+		 * @param boolean      $force_recheck Whether to force a recheck of the update status.
 		 *
-		 * @return array
+		 * @return array|object
 		 */
 		public function check_for_updates( $updates = [], $force_recheck = false ) {
 			$state = $this->get_state( $force_recheck );
 
 			$state->lastCheck      = time();
 			$state->checkedVersion = $this->get_installed_version();
 
-			// Save before checking in case something goes wrong
+			// Save before checking in case something goes wrong.
 			$this->update_state( $state );
 
 			$state->update = $this->request_update();
 
+
 			// If a null update was returned, skip to the end of the function.
 			if ( null !== $state->update ) {
+				// We treat this like an object throughout so let's convert it here.
+				$updates = (object) $updates;
+
 				// Is there an update to insert?
 				if ( version_compare( $state->update->version, $this->get_installed_version(), '>' ) ) {
-					if ( empty( $updates ) ) {
-						$updates = (object) [ 'response' => [] ];
+					if ( ! isset( $updates->response ) ) {
+						$updates->response = [];
```

Comment:
I made these changes here since the addition on line 1612 is setting `$updates` to an object.